### DISCUSSION TITLE
CharacterSpawnInfo improvement

### DISCUSSION THREAD

**Discussion by A2ur3** [FakeFishGames/Barotrauma#14679] [Code questions and modding support]
we already have spawnItemRandomly
Currently if you want to spawn a random character you need to create many items that spawn character like defensebotspawner

**Comment by A2ur3** [comment]
SpawnCharacterRandomly
MinCount
MaxCount
Probability

**Comment by jigardi** [comment]
Seems reasonable enough.
I think it would help if you gave an example of a specific usecase, and showed exactly how the difference in the XML should appear.

**Comment by A2ur3** [comment]
without `SpawnCharacterRandomly` you have to create spawner items that spawn characters and use `spawnItemRandomly`
```xml
<StatusEffect target="Character" spawnitemrandomly="true">
  <SpawnItem identifier="huskspawner" />
  <SpawnItem identifier="husk_prowlerspawner" />
  <SpawnItem identifier="husk_exosuitspawner" />
  <SpawnItem identifier="husk_chimeraspawner" />
  <RemoveCharacter />
</StatusEffect>
```
with `SpawnCharacterRandomly` you dont need to create spawner items anymore
```xml
<StatusEffect target="Character" SpawnCharacterRandomly="true">
  <SpawnCharacter speciesname="husk" />
  <SpawnCharacter speciesname="husk_prowler" />
  <SpawnCharacter speciesname="husk_exosuit" />
  <SpawnCharacter speciesname="husk_chimera" />
  <RemoveCharacter />
</StatusEffect>
```

`Commonness` - only works when SpawnCharacterRandomly="true"
<StatusEffect target="Character" SpawnCharacterRandomly="true">
  <SpawnCharacter speciesname="husk" Commonness="70" />
  <SpawnCharacter speciesname="husk_prowler" Commonness="15" />
  <SpawnCharacter speciesname="husk_exosuit" Commonness="10" />
  <SpawnCharacter speciesname="husk_chimera" Commonness="5" />
  <RemoveCharacter />
</StatusEffect>
70% chance spawns husk and 15% chance husk_prowler...
ItemSpawnInfo needs this too! otherwise you need to do this trick
<StatusEffect target="Character" SpawnCharacterRandomly="true">
  <SpawnCharacter speciesname="husk" />
  <SpawnCharacter speciesname="husk" />
  <SpawnCharacter speciesname="husk" />
  <SpawnCharacter speciesname="husk" />
  <SpawnCharacter speciesname="husk" />
  <SpawnCharacter speciesname="husk" />
  <SpawnCharacter speciesname="husk" />
  <SpawnCharacter speciesname="husk_prowler" >
  <SpawnCharacter speciesname="husk_exosuit" >
  <SpawnCharacter speciesname="husk_chimera" >
  <RemoveCharacter />
</StatusEffect>
70% chance spawns husk and 10% chance husk_prowler...

MinCount, MaxCount and Probability can be used to randomize amount
```xml
<StatusEffect target="Character">
  <SpawnCharacter speciesname="husk" MinCount="0"  MaxCount="1" />
  <RemoveCharacter />
</StatusEffect>

<StatusEffect target="Character">
  <SpawnCharacter speciesname="husk" Probability="0.5" />
  <RemoveCharacter />
</StatusEffect>
```
both status effects have 50% chance to spawn 1 husk

**Comment by A2ur3** [comment]
`multiplyafflictionsbymaxvitality`
makes `AfflictionStrength` caused by `AfflictionOnSpawn` consider character's max vitality
